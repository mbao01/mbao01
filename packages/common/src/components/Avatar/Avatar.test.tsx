import { describe, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";
import { Avatar } from "./";

describe("Avatar", () => {
  type ArgsType = Partial<{
    alt: string;
    src: string;
    ring: boolean;
    size: 4 | 8 | 12 | 16 | 24 | 32 | 48 | 64;
    shape: "round" | "circle" | "hexagon" | "triangle" | "television";
    status: "online" | "offline";
    fallback: string;
    variant:
      | "accent"
      | "default"
      | "error"
      | "ghost"
      | "info"
      | "neutral"
      | "primary"
      | "secondary"
      | "success"
      | "warning";
  }>;
  const renderAvatar = ({
    alt,
    src,
    ring,
    size,
    shape,
    status,
    variant,
    fallback,
  }: ArgsType) =>
    render(
      <Avatar size={size} status={status}>
        <Avatar.Image
          alt={alt}
          src={src}
          ring={ring}
          shape={shape}
          variant={variant}
        />
        <Avatar.Fallback
          ring={ring}
          size={size}
          shape={shape}
          variant={variant}
        >
          {fallback}
        </Avatar.Fallback>
      </Avatar>
    );

  it("renders an avatar", () => {
    const { asFragment } = renderAvatar({
      alt: "Ayomide B.",
      src: "https://github.com/shadcn.png",
      fallback: "Ayomide",
    });

    expect(screen.getByText("Ayomide")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it.each([
    "default",
    "neutral",
    "primary",
    "secondary",
    "accent",
    "ghost",
    "info",
    "success",
    "warning",
    "error",
  ] as const)("has %s variant", (variant) => {
    const fallback = `${variant} fallback`;
    const { asFragment } = renderAvatar({
      alt: "Ayomide B.",
      variant,
      src: "https://github.com/shadcn.png",
      fallback,
    });

    expect(screen.getByText(fallback)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it.each([
    { size: 4 },
    { size: 8 },
    { size: 12 },
    { size: 16 },
    { size: 32 },
    { size: 48 },
    { size: 64 },
  ] as const)("has a size ($size)", ({ size }) => {
    const fallback = `size ${size} fallback`;
    const { asFragment } = renderAvatar({
      alt: "Ayomide B.",
      src: "https://github.com/shadcn.png",
      fallback,
    });

    expect(screen.getByText(fallback)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it.each([{ status: "online" }, { status: "offline" }] as const)(
    "has $status status",
    ({ status }) => {
      const fallback = `status ${status} fallback`;
      const { asFragment } = renderAvatar({
        alt: "Ayomide B.",
        src: "https://github.com/shadcn.png",
        status,
        fallback,
      });

      expect(screen.getByText(fallback)).toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
    }
  );

  it.each([
    { shape: "round" },
    { shape: "circle" },
    { shape: "hexagon" },
    { shape: "triangle" },
    { shape: "television" },
  ] as const)("has a shape ($shape)", ({ shape }) => {
    const fallback = `${shape} shape fallback`;
    const { asFragment } = renderAvatar({
      alt: "Ayomide B.",
      src: "https://github.com/shadcn.png",
      shape,
      fallback,
    });

    expect(screen.getByText(fallback)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("has a ring", () => {
    const fallback = `Avatar with ring`;
    const { asFragment } = renderAvatar({
      alt: "Ayomide B.",
      src: "https://github.com/shadcn.png",
      ring: true,
      fallback,
    });

    expect(screen.getByText(fallback)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
