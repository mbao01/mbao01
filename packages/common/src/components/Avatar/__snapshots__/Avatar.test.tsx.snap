// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar > has 'offline' status 1`] = `
<DocumentFragment>
  <span
    class="avatar offline"
  >
    <span
      class="flex items-center justify-center w-full"
    >
      status offline fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has 'online' status 1`] = `
<DocumentFragment>
  <span
    class="avatar online"
  >
    <span
      class="flex items-center justify-center w-full"
    >
      status online fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has a ring 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full ring-3 ring-offset-base-100 ring-offset-2"
    >
      Avatar with ring
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has a shape ('circle') 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full rounded-full"
    >
      circle shape fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has a shape ('hexagon') 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full mask mask-hexagon"
    >
      hexagon shape fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has a shape ('round') 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full rounded-sm"
    >
      round shape fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has a shape ('television') 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full mask mask-squircle"
    >
      television shape fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has a shape ('triangle') 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full mask mask-triangle"
    >
      triangle shape fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has a size (4) 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full"
    >
      size 4 fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has a size (8) 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full"
    >
      size 8 fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has a size (12) 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full"
    >
      size 12 fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has a size (16) 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full"
    >
      size 16 fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has a size (32) 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full"
    >
      size 32 fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has a size (48) 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full"
    >
      size 48 fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has a size (64) 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full"
    >
      size 64 fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has accent variant 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full bg-accent text-accent-content ring-accent"
    >
      accent fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has default variant 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full bg-base-200 text-base-content ring-default"
    >
      default fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has error variant 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full bg-error text-error-content ring-error"
    >
      error fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has ghost variant 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full bg-ghost text-ghost-content ring-ghost"
    >
      ghost fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has info variant 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full bg-info text-info-content ring-info"
    >
      info fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has neutral variant 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full bg-neutral text-neutral-content ring-neutral"
    >
      neutral fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has primary variant 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full bg-primary text-primary-content ring-primary"
    >
      primary fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has secondary variant 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full bg-secondary text-secondary-content ring-secondary"
    >
      secondary fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has success variant 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full bg-success text-success-content ring-success"
    >
      success fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > has warning variant 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full bg-warning text-warning-content ring-warning"
    >
      warning fallback
    </span>
  </span>
</DocumentFragment>
`;

exports[`Avatar > renders an avatar 1`] = `
<DocumentFragment>
  <span
    class="avatar"
  >
    <span
      class="flex items-center justify-center w-full"
    >
      Ayomide
    </span>
  </span>
</DocumentFragment>
`;
